shared/provision/select-other-device/index.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/provision'
+import {ListItem2, BackButton, Box2, List, Text, Icon} from '../../common-adapters'
+import {globalMargins, styleSheetCreate, platformStyles, isMobile} from '../../styles'
+
+type Props = {|
+  devices: Array<Types.Device>,
+  onSelect: (name: string) => void,
+  onResetAccount: () => void,
+  onBack: () => void,
+  waiting: boolean,
+|}
+
+class SelectOtherDevice extends React.Component<Props> {
+  _renderItem = (index, item) => {
+    if (item === 'troubleshooting') {
+      return <Troubleshooting onResetAccount={this.props.onResetAccount} />
+    }
+
+    const {name, type} = item
+    let iconType
+    switch (type) {
+      case 'mobile':
+        iconType = 'icon-phone-32'
+        break
+      case 'desktop':
+        iconType = 'icon-computer-32'
+        break
+      case 'backup':
+        iconType = 'icon-paper-key-32'
+        break
+      default:
+        /*::
+      declare var ifFlowErrorsHereItsCauseYouDidntHandleAllTypesAbove: (type: empty) => any
+      ifFlowErrorsHereItsCauseYouDidntHandleAllTypesAbove(type);
+      */
+        iconType = 'icon-paper-key-32'
+    }
+
+    const isBackup = type === 'backup'
+
+    return (
+      <ListItem2
+        type="Small"
+        firstItem={index === 0}
+        key={name}
+        onClick={() => this.props.onSelect(name)}
+        icon={<Icon type={iconType} />}
+        body={
+          <Box2 direction="vertical" fullWidth={true}>
+            <Text type="BodySemiboldItalic">
+              {name}
+              {isBackup ? '...' : ''}
+            </Text>
+            {isBackup && <Text type="BodySmall">Paper key</Text>}
+          </Box2>
+        }
+      />
+    )
+  }
+
+  render() {
+    const items = [...this.props.devices, 'troubleshooting']
+    return (
+      <Box2
+        direction="vertical"
+        fullWidth={true}
+        fullHeight={true}
+        gap={isMobile ? undefined : 'medium'}
+        gapEnd={true}
+      >
+        <BackButton onClick={this.props.onBack} style={styles.backButton} />
+        <Box2 direction="vertical" fullWidth={true} style={styles.contentBox} gap={'medium'}>
+          <Text type={isMobile ? 'BodyBig' : 'Header'} style={styles.header}>
+            For security reasons, you need to authorize with an existing device. Which of your existing
+            devices would you like to use?
+          </Text>
+          <List
+            style={styles.list}
+            items={items}
+            renderItem={this._renderItem}
+            keyProperty="name"
```

Comment:
I see a yellow box key warning on Android on this screen -- perhaps this should be `keyProperty="key"`?

Reply:
fixed: https://github.com/keybase/client/pull/12922/commits/cf09460808bdeb99e116e2ef08b083c400c78a12